src/test/java/io/github/spark_redshift_community/spark/redshift/InMemoryS3AFileSystem.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.spark_redshift_community.spark.redshift;
+
+import java.io.*;
+import java.net.URI;
+import java.util.*;
+
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.fs.s3a.S3AFileStatus;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
+
+import org.apache.hadoop.conf.Configuration;
+
+
+public class InMemoryS3AFileSystem extends S3AFileSystem {
```

Comment:
Do you need to extend the actual `S3AFileSystem`, rather than, say `org.apache.hadoop.fs.FileSystem`? Extending a real class in test code always runs the risk of exposing some real functionality that you forgot to override.

Reply:
That's a good point, I was indeed wondering if this is hitting S3AFileSystem real methods. I'll try that.

Reply:
This works. Thanks for catching that!